### PR TITLE
Metrics on DB accesses

### PR DESCRIPTION
**Pull request by nyarly**
Some odd behavior in the DB needs more observability to track down.

### CODE REVIEW

**Comment by sous-machine-account** [comment]
LGTM

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

File: ext/storage/messages.go [hunk 1]
```diff
@@ -120,6 +136,13 @@ func (dir direction) message() string {
 	}
 }
 
+func (dir direction) String() string {
```
Comment (JC1738) [inline ext/storage/messages.go]: @nyarly do these need comments to pass linting tests?

File: ext/storage/messages.go [hunk 2]
```diff
@@ -120,6 +136,13 @@ func (dir direction) message() string {
...
+	if dir == write {
+		return "write"
+	}
+	return "read"
+}
+
 func (msg *storeMessage) EachField(fn logging.FieldReportFn) {
```
Comment (JC1738) [inline ext/storage/messages.go]: @nyarly do these need comments to pass linting tests?

**Comment by sous-machine-account** [comment]
LGTM

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Comment by lachlanh** [comment]
got a bit confused about what 'dir' was in an sql context until jim pointed out the relevant code. not specific to this pr, postgresstatemanager_write.storeManifests is starting to look large and repetitive might be a candidate for refactoring.

**Comment by lachlanh** [comment]
got a bit confused about what 'dir' was in an sql context until jim pointed out the relevant code. not specific to this pr, postgresstatemanager_write.storeManifests is starting to look large and repetitive might be a candidate for refactoring.

**Pull request merged**